3-UsingAPIs/Week1/homework/ex5-rollDiceChain.js

```diff
@@ -0,0 +1,31 @@
+// Challenge: throw five dices in sequence
+const rollDice = require('../../helpers/pokerDiceRoller');
+function rollTheDices() {
+  const results = [];
+
+  rollDice(1)
+    .then((value) => {
+      results.push(value);
+      return rollDice(2);
+    })
+    .then((value) => {
+      results.push(value);
+      return rollDice(3);
+    })
+    .then((value) => {
+      results.push(value);
+      return rollDice(4);
+    })
+    .then((value) => {
+      results.push(value);
+      return rollDice(5);
+    })
+    .then((value) => {
+      results.push(value);
+      console.log('Resolved!', results);
+    })
+    .catch((error) => console.log('Rejected!', error.message));
+}
+rollTheDices();
+// ! Do not change or remove the code below
+module.exports = rollTheDices;
```

Comment:
Your solution should work fine, but if you would like to test your understanding of using functions-returning-a-function, can you fill in the dots in this code snippet?

```js
function rollTheDices() {
  const results = [];

  const pushAndRoll = (dice) => (value) => {
    ...
  };

  rollDice(1)
    .then(pushAndRoll(2))
    .then(pushAndRoll(3))
    .then(pushAndRoll(4))
    .then(pushAndRoll(5))
    .then((value) => {
      results.push(value);
      console.log('Resolved!', results);
    })
    .catch((error) => console.log('Rejected!', error.message));
}
```

Reply:
I've doen it. Is it correct?

Reply:
It looks correct and I assume it also works as expected. But that is something for you to confirm.

Reply:
So we discussed this code as an alternative to repeating the same code for each `.then()`. Whether you prefer this DRY version over the repetitive version is a matter of personal choice.